From: John Johansen <jjohansen@suse.de>
Subject: security: Define default LSM
Patch-mainline: no
References: bnc#442668

 This patch adds a config option to select a default security module.
 Otherwise, link order determines the default LSM.

Signed-off-by: John Johansen <jjohansen@suse.de>

---
 security/Kconfig    |    9 +++++++++
 security/security.c |    2 +-
 2 files changed, 10 insertions(+), 1 deletion(-)

--- a/security/Kconfig
+++ b/security/Kconfig
@@ -60,6 +60,15 @@ config SECURITYFS
 
 	  If you are unsure how to answer this question, answer N.
 
+config SECURITY_DEFAULT
+	string "Default security module"
+	depends on SECURITY
+	default ""
+	help
+          This determines the security module used if the security=
+          boot parmater is not provided.  If a security module is not
+          specified the first module to register will be used.
+
 config SECURITY_NETWORK
 	bool "Socket and Networking Security Hooks"
 	depends on SECURITY
--- a/security/security.c
+++ b/security/security.c
@@ -18,7 +18,7 @@
 #include <linux/security.h>
 
 /* Boot-time LSM user choice */
-static __initdata char chosen_lsm[SECURITY_NAME_MAX + 1];
+static __initdata char chosen_lsm[SECURITY_NAME_MAX + 1] = CONFIG_SECURITY_DEFAULT;
 
 /* things that live in capability.c */
 extern struct security_operations default_security_ops;
